script_test.go

```diff
@@ -1768,6 +1768,52 @@ func TestWithStdout_SetsSpecifiedWriterAsStdout(t *testing.T) {
 	}
 }
 
+func TestWithEnv_UnsetsAllEnvVarsGivenEmptySlice(t *testing.T) {
+	t.Parallel()
+	want := []string{}
+
+	output, err := script.NewPipe().WithEnv(want).Exec("printenv").String()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if output != "" {
+		t.Errorf("want empty environment, got %q", output)
+	}
+}
+
+func TestWithEnvChangesExecutionEnvironment(t *testing.T) {
```

Comment:
```suggestion
func TestWithEnv_SetsGivenVariablesForSubsequentExec(t *testing.T) {
```

In case you didn't know, which by the way would be absolutely excusable, the `_` is a hint to [gotestdox](https://github.com/bitfield/gotestdox). It's a tool for printing out your test names as descriptive sentences (try it in this repo!), but without the `_` it doesn't know that `WithEnv` is a camel-cased function name and not just the words "with env".